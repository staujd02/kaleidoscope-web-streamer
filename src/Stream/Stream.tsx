import React from 'react';
import './Stream.css';

class Stream extends React.Component<StreamProps> {

    noOpMessage(){
        return "Your browser doesn't support iFrames.";
    }

    render(){
        const { source, title } = this.props.source;
        return (
            <iframe title={title} src={source}>{this.noOpMessage()}</iframe>
        );
    }

}

export default Stream;