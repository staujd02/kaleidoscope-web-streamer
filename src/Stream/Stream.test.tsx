import React from 'react';
import Stream from './Stream';
import renderer from 'react-test-renderer';

describe('The Stream', () => {

    const Source = {
        title: "Google",
        source: "https://www.google.com"
    } as Source;

    it('renders correctly', () => {
        const tree = renderer
            .create(<Stream source={Source} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});