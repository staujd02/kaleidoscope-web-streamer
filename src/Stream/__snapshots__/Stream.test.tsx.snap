// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Stream renders correctly 1`] = `
<iframe
  src="https://www.google.com"
  title="Google"
>
  Your browser doesn't support iFrames.
</iframe>
`;
